src/main/kotlin/de/itemis/mps/gradle/Common.kt

```diff
@@ -1,57 +1,80 @@
 package de.itemis.mps.gradle
 
+import de.itemis.mps.gradle.generate.GeneratePluginExtensions
 import org.apache.log4j.Logger
 import org.gradle.api.GradleException
 import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.artifacts.Configuration
-import org.gradle.initialization.GradleLauncher
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskContainer
+import java.util.Collections.emptyList
+
 import java.io.File
+import javax.inject.Inject
 
 private val logger = Logger.getLogger("de.itemis.mps.gradle.common")
 
-const val MPS_SUPPORT_MSG = "Version 1.5 doesn't only support MPS 2020.1+, please use versions 1.4 or below with older versions of MPS."
+const val MPS_SUPPORT_MSG =
+    "Version 1.5 doesn't only support MPS 2020.1+, please use versions 1.4 or below with older versions of MPS."
 
 data class Plugin(
-        var id: String,
-        var path: String
+    val id: String,
+    val path: String
 )
 
 data class Macro(
-        var name: String,
-        var value: String
+    val name: String,
+    val value: String
 )
 
-open class BasePluginExtensions {
-    var mpsConfig: Configuration? = null
-    var mpsLocation: File? = null
-    var mpsVersion: String? = null
-    var plugins: List<Plugin> = emptyList()
-    var pluginLocation: File? = null
-    var macros: List<Macro> = emptyList()
-    var projectLocation: File? = null
-    var debug = false
-    var javaExec: File? = null
+open class BasePluginExtensions @Inject constructor(of: ObjectFactory, project: Project) {
+
+    val mpsConfig: Property<Configuration> = of.property(Configuration::class.java)
+
+    val mpsLocation: RegularFileProperty = of.fileProperty().convention { File(project.buildDir, "mps") }
```

Comment:
Can inject `layout: ProjectLayout` in the constructor instead of `project` and change this property into a `DirectoryProperty` like this:
```
    val mpsLocation: DirectoryProperty = of.directoryProperty().convention(layout.buildDirectory.dir("mps"))
```

This will make it easier to work with this property later on, e.g. when computing the `plugins` or `lib` directory.